// react-mdc-web/lib/Typography/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _Typography = require('./Typography');

Object.defineProperty(exports, 'Typography', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Typography).default;
  }
});

var _Caption = require('./Caption');

Object.defineProperty(exports, 'Caption', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Caption).default;
  }
});

var _Display = require('./Display4');

Object.defineProperty(exports, 'Display4', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Display).default;
  }
});

var _Display2 = require('./Display3');

Object.defineProperty(exports, 'Display3', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Display2).default;
  }
});

var _Display3 = require('./Display2');

Object.defineProperty(exports, 'Display2', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Display3).default;
  }
});

var _Display4 = require('./Display1');

Object.defineProperty(exports, 'Display1', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Display4).default;
  }
});

var _Headline = require('./Headline');

Object.defineProperty(exports, 'Headline', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Headline).default;
  }
});

var _Title = require('./Title');

Object.defineProperty(exports, 'Title', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Title).default;
  }
});

var _Subheading = require('./Subheading2');

Object.defineProperty(exports, 'Subheading2', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Subheading).default;
  }
});

var _Subheading2 = require('./Subheading1');

Object.defineProperty(exports, 'Subheading1', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Subheading2).default;
  }
});

var _Body = require('./Body2');

Object.defineProperty(exports, 'Body2', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Body).default;
  }
});

var _Body2 = require('./Body1');

Object.defineProperty(exports, 'Body1', {
  enumerable: true,
  get: function get() {
    return _interopRequireDefault(_Body2).default;
  }
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
